llvm/test/Transforms/InstCombine/overflow_to_sat.ll

```diff
@@ -13,6 +13,18 @@ define i32 @uadd(i32 %x, i32 %y) {
   ret i32 %s
 }
 
+define i32 @uadd_cumm(i32 %x, i32 %y) {
+; CHECK-LABEL: @uadd_cumm(
+; CHECK-NEXT:    [[S:%.*]] = call i32 @llvm.uadd.sat.i32(i32 [[X:%.*]], i32 [[Y:%.*]])
+; CHECK-NEXT:    ret i32 [[S]]
+;
+  %ao = tail call { i32, i1 } @llvm.uadd.with.overflow.i32(i32 %x, i32 %y)
+  %o = extractvalue { i32, i1 } %ao, 1
+  %a = add i32 %y, %x
+  %s = select i1 %o, i32 -1, i32 %a
+  ret i32 %s
+}
```

Comment:
Please also add tests for the other intrinsics (sadd, usub, ssub).

Reply:
They already simplify without this patch, so maybe I should just only have the add functions here.

Reply:
although missing a test for `sadd` still.